### PR TITLE
Dict implementation, restoring js helper utils

### PR DESCRIPTION
**Pull request by andrzejkrawczyk**
- restored js helper utils for builting functions, isType, fixing broken builtin function sorted
- added basic Tuple implementation for **kwargs and list of tuples

### CODE REVIEW

File: batavia/utils.js [hunk 1]
```diff
@@ -684,3 +684,28 @@ batavia.make_callable = function(func) {
...
+
+/*************************************************************************
+ * utils
+ *************************************************************************/
+
+batavia.utils = batavia.utils || {};
+batavia.utils.isType = function isType(type, obj) {
+    var clas = Object.prototype.toString.call(obj).slice(8, -1);
+    return obj !== undefined && obj !== null && clas === type;
+};
```
Comment (freakboy3742) [inline batavia/utils.js]: I took these out for two reasons - firstly, they're not needed, and secondly, they're based on a string comparison, so they're not reliable anyway.

As far as I can make out, you can do reliable type checks for everything we're likely to need with `instanceof` - if there's a case where this doesn't work, let me know and I can dig into this a bit deeper.

File: batavia/core/builtins.js [hunk 2]
```diff
@@ -348,8 +348,42 @@ batavia.builtins.delattr = function(args, kwargs) {
 batavia.builtins.delattr.__doc__ = "delattr(object, name)\n\nDelete a named attribute on an object; delattr(x, 'y') is equivalent to\n``del x.y''.";
 
 batavia.builtins.dict = function(args, kwargs) {
-    throw new batavia.builtins.NotImplementedError(
-        "Builtin Batavia function 'dict' not implemented");
+    if (args.length > 1) {
```
Comment (freakboy3742) [inline batavia/core/builtins.js]: The top of builtins.js contains a template for builtin functions to follow; there's a couple of edge cases here that you haven't covered.

Comment (andrzejkrawczyk) [inline batavia/core/builtins.js]: isType is used in builtin sorted, taking that out will break the function.

Also instanceof is good enough only for checking custom types with custom constructors.
If you need to check JavaScript type this is totally broken. 
More about this with examples in this article:
http://bonsaiden.github.io/JavaScript-Garden/#types.equality

Should i Still remove isType util and rework sorted function in builtins?

File: batavia/utils.js (hunk 1 above)
Comment (andrzejkrawczyk) [inline batavia/utils.js]: isType is used in builtin sorted, taking that out will break the function.

Also instanceof is good enough only for checking custom types with custom constructors.
If you need to check JavaScript type this is totally broken. 
More about this with examples in this article:
http://bonsaiden.github.io/JavaScript-Garden/#types.equality

Should i Still remove isType util and rework sorted function in builtins?

**Comment by freakboy3742** [comment]
Closing this PR - #50 implemented dict().

**Pull request closed without merging**